CMakeLists.txt

```diff
@@ -123,6 +133,70 @@ if(USE_GPU)
     ADD_DEFINITIONS(-DUSE_GPU)
 endif(USE_GPU)
 
+if(USE_CUDA)
+    find_package(CUDA REQUIRED)
+    include_directories(${CUDA_INCLUDE_DIRS})
+    LIST(APPEND CMAKE_CUDA_FLAGS -g -Xcompiler=-fopenmp -Xcompiler=-fPIC -Xcompiler=-Wall -lineinfo)
```

Comment:
I guess the current solution works only because the default value of `USE_OPENMP` is `True`. For `False` `${OpenMP_CXX_FLAGS}` will be undefined because `find_package(OpenMP REQUIRED)` is called later in the code.
I propose to move 
```
SET(USE_OPENMP ON CACHE BOOL "CUDA requires OpenMP" FORCE)
```
line to the first `if(USE_CUDA)` branch with project definition: https://github.com/microsoft/LightGBM/pull/3160#discussion_r470853440. It will allow to successfully compile the project if one pass `-DUSE_OPENMP=FALSE` or fail with meaningful error about that OpenMP is required for CUDA version.

Reply:
Adding this section

```
+if(USE_CUDA)
+    SET(USE_OPENMP ON CACHE BOOL "CUDA requires OpenMP" FORCE)
+endif(USE_CUDA)
+
 if(USE_OPENMP)
```